src/twiddle/hash/minhash.c

```diff
@@ -136,23 +140,24 @@ float tw_minhash_estimate(const struct tw_minhash *a,
 
   uint32_t n_registers_eq = 0;
 
-#define MINH_EST_LOOP(simd_t, simd_load, simd_cmpeq, simd_maskmove, eq_mask)   \
+#define MINH_EST_LOOP(simd_t, simd_load, simd_cmpeq, simd_add, simd_set1)      \
   const size_t n_vectors =                                                     \
       n_registers * TW_BYTES_PER_MINHASH_REGISTER / sizeof(simd_t);            \
+  simd_t acc = simd_set1(0);                                                   \
   for (size_t i = 0; i < n_vectors; ++i) {                                     \
     simd_t *a_addr = (simd_t *)a->registers + i,                               \
            *b_addr = (simd_t *)b->registers + i;                               \
-    const simd_t v_cmp = simd_cmpeq(simd_load(a_addr), simd_load(b_addr));     \
-    const int h_cmp = simd_maskmove(v_cmp);                                    \
-    n_registers_eq += __builtin_popcount(h_cmp & eq_mask);                     \
-  }
+    acc = simd_add(acc, simd_cmpeq(*a_addr, *b_addr));                         \
+  }                                                                            \
+  for(size_t i = 0; i < sizeof(simd_t)/sizeof(uint32_t); i++)	               \
+    n_registers_eq -= ((uint32_t *)&acc)[i];
```

Comment:
`-=`?

Reply:
These are sums of 0's and -1's, so I subtract the subtotals here.  I tried _mm_sub_epi32 to get positive subtotals, but it was slower somehow.

Reply:
That's surprising. I'll try to reproduce.

Reply:
Somehow forgot that cmpeq returned 0xFFFFFF.